api/server.go

```diff
@@ -120,53 +165,65 @@ func getAmount(r *http.Request) (int, error) {
 }
 
 func (a *API) addUserHandler(w http.ResponseWriter, r *http.Request) {
-	amount, err := getAmount(r)
-	if amount <= 0 || err != nil {
-		msg := fmt.Sprintf("invalid amount: %s", r.FormValue("amount"))
-		writeJsonResponse(w, http.StatusBadRequest, map[string]interface{}{"error": msg})
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
 		return
 	}
 
-	lt, err := a.getLoadTestById(w, r)
-	if err != nil {
+	amount, err := getAmount(r)
+	if amount <= 0 || err != nil {
+		writeResponse(w, http.StatusBadRequest, &APIResponse{
+			Status: lt.Status(),
+			Error:  fmt.Errorf("invalid amount: %s", r.FormValue("amount")),
+		})
 		return
 	}
 
 	i := 0
-	var addError string
+	var addError error
 	for ; i < amount; i++ {
 		if err := lt.AddUser(); err != nil {
-			addError = err.Error()
+			addError = err
 			break // stop on first error, result is reported as part of status
 		}
 	}
 
-	writeJsonResponse(w, http.StatusOK, map[string]interface{}{"message": fmt.Sprintf("%d users added", i), "error": addError, "status": lt.Status()})
+	writeResponse(w, http.StatusOK, &APIResponse{
```

Comment:
Here (and in `removeUserHandler`) we end up returning `http.StatusOK` even when `lt.AddUser()` fails. 
It might be okay since we specify an error object but if so I'd like to be consistent and do the same in `runLoadAgentHandler` and `stopLoadAgentHandler`.

Reply:
Can you expand what you meant by "do the same" ? AFAI see we are indeed returning statusok everywhere.

Reply:
If Stop fails we are still returning `500`.